backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/VerkuendungsImportService.java

```diff
@@ -141,9 +141,20 @@ public VerkuendungImportProcess getStatus(
   }
 
   @Override
-  public void processNormendokumentationspaket(
+  public Verkuendung processNormendokumentationspaket(
```

Comment:
Why using the same `ProcessNormendokumentationspaketUseCase` but then doing a switch and calling different methods? For me it feels like two different use cases, one is the background process that takes the UUID and the other one is the uploading process which does not work with an UUID but with the ZIP directly. Also specially because the upload needs the _Verkuendung_ returned but the background process does not.

Reply:
That was my first idea as well, but as i couldn't find a naming that properly explains the difference between the two I got more convinced that it really is the same use case with just different arguments for how to find the data to work on.

Reply:
For me personally felt quite confusing, specially because completely different Options, different return values and different processing (no signature validation). The only thing they share is the parsing, validation and saving, yeah big part of the functionality but not all. But it is just my taste, I don't expect you to separate it now into two different use cases :)